cgd_tx_eff/src/edu/ohsu/compbio/txeff/util/find_genotype_in_avinput.py

```diff
@@ -0,0 +1,138 @@
+'''
+Created on Sep. 27, 2022
+This script reads a bunch of genotypes from one file, looks those genotypes up in an existing annovar input file, 
+and writes each of those annovar lines to a new file.
+@author: pleyte
+'''
+
+import argparse
+import logging
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+stream_handler = logging.StreamHandler()
+logging_format = '%(levelname)s: [%(filename)s:%(lineno)s - %(funcName)s()]: %(message)s'
+
+stream_format = logging.Formatter(logging_format)
+stream_handler.setFormatter(stream_format)
+stream_handler.setLevel(logging.DEBUG)
+logger.addHandler(stream_handler)
+
+class Genotype(object):
+    def __init__(self, genotype:str):
+        parts = genotype.split('-', 4)
+        self.chromosome = parts[0]
+        self.position = parts[1]
+        self.reference = parts[2]
+        self.alt = parts[3]        
+        
+    def __str__(self, *args, **kwargs):
+        return self.chromosome + '-' + self.position + '-' + self.reference + '-' + self.alt
+    
+class FindGenotypeInAvinput(object):
+    '''
+    classdocs
+    '''
+    def __init__(self, genotype_file, av_input_file, av_output_file):
+        '''
+        Constructor
+        '''
+        self.genotype_file = genotype_file
+        self.av_input_file = av_input_file
+        self.av_output_file = av_output_file
+    
+    def _get_genotypes(self):
+        '''
+        Return a list of genotype object read from the input file
+        '''
+        logger.info(f"Reading genotypes from {self.genotype_file}")
+        genotypes = list()
+        with open(self.genotype_file,"r") as file:
+            for line in file:                
+                genotypes.append(Genotype(line.strip()))
+        
+        return genotypes
+    
+    def _is_line_matches_genotype(self, tab_delimited_line, genotype: Genotype):        
+        '''
+        return true if the tab delimiited line defines the genotype
+        '''        
+        # Split the tab delimited line into multiple parts 
+        line_parts = tab_delimited_line.split("\t")
+        if(line_parts[0] == 'chr'+genotype.chromosome and 
+           line_parts[1] == genotype.position and 
+           line_parts[3] == genotype.reference and 
+           line_parts[4] == genotype.alt):
+            return True
+        else:
+            return False
+                
+    def _get_matching_avinput(self, genotypes):
+        '''
+        '''
+        logger.info(f"Attempting to find {len(genotypes)} genotypes in {self.av_input_file}")
+        annovar_lines = list()
+        
+        logger.info(f"There are {len(genotypes)} genotypes")
+        
+        with open(self.av_input_file, "r") as file:
+            for line in file:
+                for genotype in genotypes:
```

Comment:
could make this "for genotype" loop in to its own function to avoid using break, but not a big deal